js/components/utils.js

```diff
@@ -632,43 +632,95 @@ var Utils = {
-      for (var i = 0; i < localization.availableLanguageFile.length; i++) {
-        availableLanguages = localization.availableLanguageFile[i];
-        // if lang file available set lang to browser primary lang
-        if (primaryLanguage === availableLanguages.language) {
-          language = primaryLanguage;
-        }
-      }
+  /**
+   *
+   * @param {Object} localization - location configuration object
+   * @param {String} localization.defaultLanguage - value for default language
+   * @returns {String} defaultLanguage - language code from file "skin.json" (
+   * before executing function "onSkinMetaDataFetched") or language code that was set
+   * in the page-level parameters (after executing function "onSkinMetaDataFetched")
+   */
+  getDefaultLanguage: function(localization) {
+    let defaultLanguage = '';
+    if ( (typeof localization === "object") && (localization !== null) ) {
+      defaultLanguage = localization.defaultLanguage ?
+        localization.defaultLanguage
+      : defaultLanguage;
     }
+    return defaultLanguage;
+  },
 
-    return language;
+  /**
+   *
+   * @returns {String} two-digit value of an user's system language or an empty string
+   */
+  getUserBrowserLanguage: function() {
+    // Examples of valid language codes are: "en", "en-US", "de", "fr", etc.
+    let navigator = window.navigator;
+    let language;
+    if (navigator) {
+      language = (
+        navigator.language ||
+        // "language" property returns the language of the browser application in
+        // Firefox, Opera, Google Chrome and Safari
+        navigator.userLanguage ||
+        // "userLanguage" property returns the current Regional and Language settings
+        // of the operating system in
+        // Internet Explorer and the language of the browser application in Opera
+        navigator.systemLanguage
+        // "systemLanguage" property returns the language edition of the operating system in
+        // Internet Explorer
+      )
+    }
+    return language ? language.substr(0, 2).toLowerCase() : ''; // remove lang sub-code
+  },
+
+  /**
+   *
+   * @param {Array} availableLanguageList - array of objects of language code values
+   * with the key "language"
+   * @param languageCode - two-digit value of the language
+   * @returns {Boolean} "true" if the corresponding language code "languageCode" is
+   * in the array of available languages "availableLanguageList",
+   * otherwise "false"
+   */
+  isLanguageCodeInAvailablelLanguageFile: function(availableLanguageList, languageCode) {
+    if ( !(availableLanguageList && Array.isArray(availableLanguageList) && languageCode) ) {
+      return false;
+    }
+    for (let index = availableLanguageList.length - 1; index >= 0; index--) {
+      let availableLanguageCode = availableLanguageList[index].language;
+      if (languageCode === availableLanguageCode) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
```suggestion for lines 717 - 723
    return !!availableLanguageList.find(languageObj => languageObj.language === languageCode);
```
This is not just simply shorter, but eliminates also return from an iteration of loop that is a sort of antipattern.